### PR TITLE
Kavitha, Ashish | IPD diagnosis Display control| BAH-3280|1205516692048403

### PR DESCRIPTION
**Pull request by kavitha-sundararajan**
- adds diagnosis display control for the new IPD dashboard
JIRA ref : [BAH-3280](https://bahmni.atlassian.net/browse/BAH-3280)
Sample screenshot:
Screenshot 2023-10-30 at 3 32 12 PM

### CODE REVIEW

File: src/components/Dashboard/Dashboard.spec.jsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { act, render, fireEvent, screen } from "@testing-library/react";
 
 import Dashboard from "./Dashboard";
 
-describe("Dashboard", () => {
+describe.skip("Dashboard", () => {
```
Comment (ayush34957) [inline src/components/Dashboard/Dashboard.spec.jsx:6]: we can remove it now... identified the issue why the github test action was failing

File: public/config.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (ayush34957) [inline public/config.json:1]: we have moved this file location and is not needed

File: src/features/DisplayControls/Diagnosis/components/Diagnosis.jsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    const[isLoading, setIsLoading] = useState(true);
+
+    const NoDiagnosisMessage = (
+    <FormattedMessage
+        id={"NO_DIAGNOSIS_MESSAGE"}
+        defaultMessage={"No Diagnosis found for this patient"}
+    />
+    );
+  
+    const diagnosisHeaders = [
```
Comment (ayush34957) [inline src/features/DisplayControls/Diagnosis/components/Diagnosis.jsx:31]: can we move the header config to Diagnosis utility file?

File: src/features/DisplayControls/Diagnosis/components/Diagnosis.jsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        header: 'Diagnosis Date ',
+        key: 'diagnosisDate',
+        isSortable: true 
+    },
+    ];
+
+    const mapDiagnosisData = (diagnosisList) => {
+        const mappedDiagnoses = diagnosisList === undefined ? [] : diagnosisList.map((diagnosis) => {
+            let status = diagnosis.diagnosisStatusConcept ? "Inactive" : "Active";
+            let diagnosisDate = new Date(diagnosis.diagnosisDateTime).toLocaleDateString();
```
Comment (ayush34957) [inline src/features/DisplayControls/Diagnosis/components/Diagnosis.jsx:73]: what is the date format? I think we are following DD MMM YYYY, please check and if that's the case create a common utility method in utils for formatting date and time and use it in here.

File: src/features/DisplayControls/Diagnosis/styles/Diagnosis.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.no-dignosis-message {
+  padding-top: 0.625rem;
```
Comment (ayush34957) [inline src/features/DisplayControls/Diagnosis/styles/Diagnosis.scss:2]: it's always better to use px or percentage for margin and padding depending on the use case

File: src/components/Dashboard/Dashboard.spec.jsx (hunk 1 above)
Comment (kavitha-sundararajan) [reply src/components/Dashboard/Dashboard.spec.jsx]: removed skip

File: src/features/DisplayControls/Diagnosis/components/Diagnosis.jsx (hunk 3 above)
Comment (kavitha-sundararajan) [reply src/features/DisplayControls/Diagnosis/components/Diagnosis.jsx]: removed the file

File: src/features/DisplayControls/Diagnosis/components/Diagnosis.jsx (hunk 4 above)
Comment (kavitha-sundararajan) [reply src/features/DisplayControls/Diagnosis/components/Diagnosis.jsx]: added a date util file

File: src/features/DisplayControls/Diagnosis/styles/Diagnosis.scss (hunk 5 above)
Comment (kavitha-sundararajan) [reply src/features/DisplayControls/Diagnosis/styles/Diagnosis.scss]: updated to px

File: public/config.json (hunk 2 above)
Comment (kavitha-sundararajan) [reply public/config.json]: removed the file

**Review by ayush34957 — APPROVED** [review]

**Pull request merged**